File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_water = INT_MIN;
        int n = height.size();
        int left = 0, right = n-1;
        while(left < right) {
            int w = right - left;
            int h = min(height[left],height[right]);
            max_water = max(max_water,w*h);
            if (height[left] < height[right])
                left++;
            else
                right--;
        }
        return max_water;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int n = height.length;
        int i = 0, j = n-1;
        int max_water = 0;
        while(i < j) {
            int water = (j-i) * Math.min(height[i],height[j]);
            max_water = Math.max(max_water,water);
            if(height[i] < height[j]) i++;
            else j--;
        }
        return max_water;
    }
}